tests/test_fluent.py

```diff
@@ -409,3 +410,369 @@ def test_meta(self):
+                        ],
+                        (CatchallKey("other"), CatchallKey("masculine")): [
+                            "There are many emails for him"
+                        ],
+                    },
+                ),
+                comment="Multiple selectors",
+            ),
+            Entry(
+                id=["-term"],
+                value=PatternMessage(["Term"]),
+                comment="Term",
+            ),
+            Entry(
+                id=["term-reference"],
+                value=PatternMessage(
+                    [
+                        "Term ",
+                        Expression("-term", FunctionAnnotation("message")),
+                        " Reference",
+                    ],
+                ),
+                comment="TermReference",
+            ),
+            Entry(
+                id=["string-expression"],
+                value=PatternMessage([Expression("")]),
+                comment="StringExpression",
+            ),
+            Entry(
+                id=["number-expression"],
+                value=PatternMessage(
+                    [Expression("5", FunctionAnnotation("number"))],
+                ),
+                comment="NumberExpression",
+            ),
+            Entry(
+                id=["attribute-expression"],
+                value=PatternMessage(
+                    [Expression("my_id.title", FunctionAnnotation("message"))],
+                ),
+                comment="MessageReference with attribute (was: AttributeExpression)",
+            ),
+            Entry(
+                id=["selector-nested"],
+                value=SelectMessage(
+                    selectors=[
+                        Expression(VariableRef("gender"), FunctionAnnotation("string")),
+                        Expression(VariableRef("num"), FunctionAnnotation("number")),
+                    ],
+                    variants={
+                        ("masculine", "one"): ["There is one email for him"],
+                        ("masculine", CatchallKey("other")): [
+                            "There are many emails for him"
+                        ],
+                        (CatchallKey("feminine"), "one"): [
+                            "There is one email for her"
+                        ],
+                        (
+                            CatchallKey("feminine"),
+                            CatchallKey("other"),
+                        ): ["There are many emails for her"],
+                    },
+                ),
+                comment="Nested selectors",
+            ),
+        ]
+        self.assertEqual(res, Resource(sections=[Section(id=[], entries=entries)]))
+        self.assertEqual(
+            "".join(fluent_serialize(res)),
+            dedent(
+                """\
+                # Simple string
+                title = About Localization
+                # Multiline string: press Shift + Enter to insert new line
```

Comment:
Can we preserve newlines?

I worry about the diff size.

Reply:
If you mean newlines e.g. before this comment, then no, we can't. They are not relevant for the resource contents, and therefore are left out of the data model.

And actually, doing so (and otherwise normalising the syntaxes) will minimize diff sizes. The key thing to keep in mind here is that in practice we don't have anywhere where both humans _and_ automation need to both manage contents.

Reply:
SGTM. Let's keep PMs in the loop about this.